testing/ostest/ostest_main.c

```diff
@@ -680,9 +715,19 @@ int main(int argc, FAR char **argv)
 
   /* Verify that we can spawn a new task */
 
+#ifdef CONFIG_BUILD_KERNEL
+  posix_spawnattr_init(&attr);
+  param.sched_priority = PRIORITY;
+  posix_spawnattr_setschedparam(&attr, &param);
+  posix_spawnattr_setstacksize(&attr, STACKSIZE);
+  status = posix_spawn(&result, arg[0], NULL, &attr,
+                       (char * const *)arg, NULL);
+  if (status != 0)
+#else
   result = task_create("ostest", PRIORITY, STACKSIZE, user_main,
```

Comment:
why not always use pthread_create

Reply:
Subtask user_main need check the argc and argv list, but no copy argv list in pthread_create for parameter. If need, should rework for user_main, but do not change anything if posix_spawn in kernel mode.
static int user_main(int argc, char *argv[])